sklearn/linear_model/tests/test_ridge.py

```diff
@@ -2265,5 +2265,21 @@ def test_metadata_routing_with_default_scoring(metaestimator):
     metaestimator().get_metadata_routing()
 
 
+@pytest.mark.usefixtures("enable_slep006")
+@pytest.mark.parametrize(
+    "metaestimator, make_dataset",
+    [
+        (RidgeCV(), make_regression),
+        (RidgeClassifierCV(), make_classification),
+    ],
+)
+def test_set_score_request_with_default_scoring(metaestimator, make_dataset):
+    """Test that `set_score_request` is set within `RidgeCV.fit()` and
+    `RidgeClassifierCV.fit()` when using the default scoring. Regression test for the
+    fix in PR #29634."""
+    X, y = make_dataset(n_samples=100, n_features=5, random_state=42)
+    metaestimator.fit(X, y, sample_weight=np.ones(X.shape[0]))
+
```

Comment:
Is this test complete? It doesn't seem to be doing what it says it does. We should check that sample_weight is passed to the scorer. Maybe compare with and w/o passing sample_weight here?

Reply:
Will comparing with sample_weight being all ones be suitable? If we want to compare I think we should have some different values for the samples and then maybe confirm that the results differ? Or we could set some samples with a weight of 2 and then duplicate those samples in the original data and compare that the results match. Like we do in other cases as well?

Reply:
I cannot see what you mean?

I think this test does what it is supposed to do: If we would call fit() and `set_score_request` was not hard-codedly set within `fit()` (as it is here in `_get_scorer()`, called by `fit()`) we would get the following Error: 

```
sklearn.exceptions.UnsetMetadataPassedError: [sample_weight] are passed but are not explicitly set as requested or not requested for RidgeClassifierCV.score, which is used within RidgeClassifierCV.fit. Call `RidgeClassifierCV.set_sco...
```

The test makes sure that no error is raised.

Reply:
If the test is not doing what the description claims, then we can also change the description.
The routing itself is probably tested somewhere else already. Here, we are only fixing two bugs.

Reply:
In other words, in case this is not clear, this test is testing [this commit](https://github.com/scikit-learn/scikit-learn/pull/29634/commits/3956f2cdacccfbd9bcabd3a9ff43ca935721ccdf) of Adrin.

Reply:
I see. Maybe the description of the test can be updated a bit to indicate that this the test verifies that no error is raised now for the specific case where it was raising previously?